### DISCUSSION TITLE
old code with newer versions?

### DISCUSSION THREAD

**Discussion by pr-apes** [speedata/publisher#579] [Q&A] [answered]
@pgundlach,

I use Publisher mainly for already written scripts or recipes (for some predefined manipulations with PDF documents).[^scripts]

After 4.17.0, I have been experiencing issues with the XPath parser (both `luxor` and `lxpath`).

Which would be the way to test XPath expressions so that they would be understood by `lxpath`?

Many thanks for your help.

[^scripts]: They are invoked by dropping them in `.bat` files, which invoke Publisher (sometimes using filters).

**Comment by pgundlach** [comment, marked as answer]
Tough question. If you can share your (problematic) code, I can help debugging. But I don't any sensible way to test these expression.

lxpath is based on https://github.com/speedata/lxpath and I have (for my tests) a small main.lua file that invokes the lxpath parser and prints out the result, but since you need a special format for the XML file, it is not usable for non trivial test cases...

**Reply by pr-apes** [reply]
The new version seems to have (at least one) less issue with old code.

I try to isolate the offending code and share it with you.

The most problematic part were mainly conditionals.

I will check the old code with the new version and new parser and lets see what happens.

Many thanks for your help.

**Reply by pgundlach** [reply]
You can always send code to my email address if you don't want to share it to the public. See https://www.speedata.de/imprint/

**Reply by pr-apes** [reply]
@pgundlach,

many thanks for your offer, but my problem is to isolate the issue in minimal samples.

Sharing that minimal code in public shouldn't be an issue (elaborating the report is what takes time).

Many thanks for your help.
